controller/app/models/application.rb

```diff
@@ -1735,7 +1735,8 @@ def process_commands(result_io, component_id=nil, gear=nil)
         if gear
           pi = PortInterface.create_port_interface(gear, component_id, *command_item[:args])
           gear.port_interfaces.push(pi)
-          pi.publish_endpoint(self)
+          public_ip = command_item[:args][0]["external_address"] || gear.get_public_ip_address
+          pi.publish_endpoint(self, public_ip)
```

Comment:
@Miciah this change, as well as the change to port_interface.rb, are necessary due to the way the PortInterface class gets the external "public" ip address of the node the gear is on. When calling PortInterface.external_address, the gear proxy is used and will return the old ip address of the moved gear.

The correct external ip address is passed in the NOTIFY_ENDPOINT_CREATE arguments, so we use that instead if its available, similar to the way the external ip address is handled for NOTIFY_ENDPOINT_DELETE events.